src/S3/Input/CopyObjectRequest.php

```diff
@@ -859,4 +861,17 @@ public function requestUri(): string
 
         return "/{$uri['Bucket']}/{$uri['Key']}";
     }
+
+    public function validate(): void
+    {
+        foreach (['Bucket', 'CopySource', 'Key'] as $name) {
+            if (null === $value = $this->$name) {
+                throw new InvalidArgument(sprintf('Missing parameter "%s" when validating the "%s". The value cannot be null.', $name, __CLASS__));
```

Comment:
Shouldn't we add the className? When validating a sub-object, the name of the parameter could be meaningless

Reply:
We are using `__CLASS__ `. Is there need for something more?

Reply:
nvm.. sorry for the noise